docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,328 @@
+
+#### Hard to annotate an existing multi-arch image
+
+A security engineer Cindy needs to use image lifecycle annotations with ORAS to mark when the vulnerable image should be considered end of life (EoL) and no longer used by dependent services. 
+However, as there are multi-arch images and separate arch-specific images maintained by service teams, it is cumbersome that they can only apply annotations manually to each arch-specific image separately instead of the multi-arch image. Most consumers of multi-arch images only interact with the parent image index for tagging and deployment. The EoL annotation is not available on the multi-arch image (index), which makes the multi-arch image (index) unverifiable.
+
+![separate-annotation](./img/separate-annotation.png)
+
+#### Hard to inspect a multi-arch image locally
+
+It’s hard to inspect a multi-arch image’s manifest content locally since docker buildx stores the multi-arch image in Docker’s build cache. David is a sofware engineer who needs to tag and push the multi-arch image to a registry after build. He also wants to inspect the multi-arch image locally before the release. It makes the multi-arch image a black box, which is hard for local testing.
+
+## Proposed solution
+
+Ideally, if ORAS extends the ability to manage a multi-arch image by introducing create, update, and annotate a multi-arch image, the problems listed above can be resolved. Managing an image index for a multi-arch image should be as easy as playing Legos. 
+
+The proposed CLI commands for managing a multi-arch image are listed below. The detailed use case and subcommands are articulated in the CLI Spec section.
+
+- Create a multi-arch image using OCI image index: `oras manifest index create`
+- Update a multi-arch image using OCI image index: `oras manifest index update`
+- Inspect a multi-arch image using OCI image index: `oras manifest inspect`
+- Annotate a multi-arch image using OCI image index: `oras manifest create/update --annotation`
+- List platform information of tags in a repository: `oras repo tags`
+
+The proposal uses an OCI image layout as the local store to create a multi-arch image as an OCI image index locally, then push the multi-arch image to the remote registry with ORAS.  
+
+### User scenario and desired experience
+
+The sample workflow will be as following:
+
+1. Assume there are two arch-specific images tagged as _“v1-linux-amd64”_ and _“v1-linux-arm64” and have been pushed to an OCI image layout _layout-dir_. List the tags in the OCI image layout:
+
+```bash
+$ oras repo tags --oci-layout layout-dir 
+
+v1-linux-amd64 
+v1-linux-arm64 
+```
+
+2. Create a multi-arch image by composing two image manifests into an image index, tag it with “v1” and push the image index to an OCI image layout “layout-dir” automatically:  
+
+```
+$ oras manifest index create --oci-layout layout-dir:v1 v1-linux-amd64 v1-linux-arm64 
+
+Resolving  v1-linux-amd64 
+Resolved   v1-linux-amd64 
```

Comment:
Is `Resolve` a good verb to use here? We currently use the verb `Fetch`. I think `resolve` sounds better to the user, but `fetch` is more accurate about what is done. What do you think? @qweeah

Reply:
Updated the output to fetch and add a status line